### PR TITLE
[FIX] website: don't error on inaccessible model

### PR DESCRIPTION
**Pull request by xmo-odoo**
When deleting a page, `search_url_dependencies` tries to trawl through all models to see if they might have a link to the page being deleted.

However if the user invoking that function does not have access to a model with an HTML field, it raises an error. Given pages are managed by website designers which are *not administrators* there is no reason to believe the current user has access to every model in the database (not that even admins do these days). Since `search_url_dependencies` is a best-effort search anyway, just ignore any model to which the current user doesn't have access.

An alternative would be to do the search in sudo mode, but that doesn't seem necessary, and could even be problematic if a match is found:

- it might leak information the user should not access (because the record name is returned, as well as the model & field names)
- it will trigger further access errors (because links to problematic records are provided, on which the user might want to click)

### CODE REVIEW

**Review by rdeodoo — CHANGES_REQUESTED** [review]
@robodoo r-

> Since `search_url_dependencies` is a best-effort search anyway

It's not, at least it's not meant to be.
That's partly why we did https://github.com/odoo/odoo/commit/15c350ae47332d28f64f6a552b5c5d6930054e7d
The goal of this is to be reliable and to be sure that if the system doesn't miss anything, you are safe to edit or delete the link without having to modify something else, at least internally on your website.

Note that this introduced some perf issues that we are going to fix with another PR.
My point was detailed at https://github.com/odoo/odoo/pull/116839#issuecomment-1516011520
> tl;dr: feature should ideally remain as is for small and medium companies (search everywhere, there won't be many records, exclude tables when needed like mail.message, **ensure we provide a reliable warning**) but **should be tweaked for big ones** to not impact perfs (do search_count first, if more than X records -> Don't do the search, **just warn that we couldn't provide complete search due to heavy amount of records** and link might be used somewhere -> **but still do the search for pages/views/menus as it's the core feature**).

I am not against re-thinking the feature but this PR right now would go against what we tried and still try to achieve: either have reliable result and if not (too many records / big DB (rare case), could also include your right error case) at least let the user know that there might be missed results or incomplete results.

Depending of what we come up with as a warning message for the perf issue, we might do the same with this right issue.
But IMHO, I'd just sudo it, we should've used sudo at first implementation.

**Comment by rdeodoo** [comment]
> Then you might want to fix the docstring:

Correct, sorry about that, it's misleading. It should've been updated with the previous "refactoring".

> search will only returns the records to which the current user has access. [..] sets active_test before calling search_url_dependencies inactive records are also ignored already.

I will fix that, it's not coherent with what we tried to achieve indeed.

> "Just sudo it" is broken for the reasons explained, the design leaks information and leads to further access errors as the user will be given links to records they can not access.

I will see what we can do, I don't think it's that much of an issue to show an unreachable link (because you don't have access), but we can surely do something about it like detecting those "403" links and making it so it's not a link but plain text with a warning about the access restriction.

What we are trying to achieve is to not have people delete a link (or an image as we have the same behavior on the media dialog for images) without noticing that it left a dead link or a broken image on another page/blog/product/whatever without him ever noticing, and that would really ruin those pages (probably even worst for images).

This seems important to me, but maybe it's overkill.
Apart from the technical aspect, I guess it's really more a functional decision :shrug: 
What do you think @pierre-pde ?

**Comment by pierre-pde** [comment]
@rdeodoo  & @xmo-odoo ,
I think this feature is nice, and we should aim to have it as precise as possible.
I agree with Romain's proposition. In case the thing the user is trying to delete is used in a object that he doesn't have access to, then we should show a message like:
"Warning, the IMAGE you want to delete is used in a BLOG POST to which you don't have access".
It's not ideal but still better than no warning at all.

**Comment by rdeodoo** [comment]
The missing `sudo` and missing `active_context=False` will be handled in the 16.1 forward port of https://github.com/odoo/odoo/pull/116839 which is adding the kinda same checks for image deletion from the website builder.
It is coming with the "max 1000 records" SME/big company different mechanism which will also be applied to this 16.1 website.menu dependencies.

The mentioned PR will use the already existing utils in its 15.0 forward port.

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

Revived this one but retargeted 16.0.
See https://github.com/odoo/odoo/pull/116839#issuecomment-1625566749, we are not going to do any change in stable anymore as we will be coming up with a full refactoring of the feature in master with a "precise/exact and no perf impact" solution.
That solution will correctly handle the "not access" right without leaking information like the name (which is already leaked in Odoo when you access a record in the 403 modal but still).
